controller/src/main/java/io/pravega/controller/server/rpc/auth/AuthContext.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.server.rpc.auth;
+
+import io.pravega.auth.AuthHandler;
+import lombok.Getter;
+
+import java.security.Principal;
+
+/**
+ * A gateway to objects held in gRPC {@link io.grpc.Context} by the {@link AuthInterceptor}, for the current gRPC
+ * request.
+ */
+public class AuthContext {
+
+    @Getter
+    private Principal principal;
+
+    @Getter
+    private AuthHandler authHandler;
+
+    private AuthContext(Principal principal, AuthHandler authHandler) {
```

Comment:
`@NonNull` for both arguments or explain in your JavaDoc which arguments can be null.

Reply:
Nulls are allowed for both the params. I've added JavaDocs comments to express that, as suggested.